contracts/votes/src/storage.rs

```diff
@@ -0,0 +1,295 @@
+    pub timestamp: u64,
+}
+
+//********** Storage Utils **********//
+
+/// Bump the instance lifetime by the defined amount
+pub fn extend_instance(e: &Env) {
+    e.storage()
+        .instance()
+        .extend_ttl(INSTANCE_LIFETIME_THRESHOLD, INSTANCE_BUMP_AMOUNT);
+}
+
+/// Fetch an entry in persistent storage that has a default value if it doesn't exist
+fn get_persistent_default<K: IntoVal<Env, Val>, V: TryFromVal<Env, Val>, F: FnOnce() -> V>(
+    e: &Env,
+    key: &K,
+    default: F,
+    bump_threshold: u32,
+    bump_amount: u32,
+) -> V {
+    if let Some(result) = e.storage().persistent().get::<K, V>(key) {
+        e.storage()
+            .persistent()
+            .extend_ttl(key, bump_threshold, bump_amount);
+        result
+    } else {
+        default()
+    }
+}
+
+/// Fetch an entry in temporary storage that has a default value if it doesn't exist
+fn get_temporary_default<K: IntoVal<Env, Val>, V: TryFromVal<Env, Val>, F: FnOnce() -> V>(
+    e: &Env,
+    key: &K,
+    default: F,
+    ledgers_to_live: u32,
+) -> V {
+    if let Some(result) = e.storage().temporary().get::<K, V>(key) {
+        e.storage()
+            .temporary()
+            .extend_ttl(key, ledgers_to_live, ledgers_to_live);
+        result
+    } else {
+        default()
+    }
+}
+
+//********** Instance **********//
+
+// Initialization flag
+
+/// Check if the contract has been initialized
+pub fn get_is_init(e: &Env) -> bool {
+    e.storage().instance().has(&IS_INIT_KEY)
+}
+
+/// Set the contract as initialized
+pub fn set_is_init(e: &Env) {
+    e.storage()
+        .instance()
+        .set::<Symbol, bool>(&IS_INIT_KEY, &true);
+}
+
+// Token
+
+pub fn get_token(e: &Env) -> Address {
+    e.storage().instance().get(&TOKEN_KEY).unwrap_optimized()
+}
+
+pub fn set_token(e: &Env, address: &Address) {
+    e.storage().instance().set(&TOKEN_KEY, address);
+}
+
+// Metadata
+
+pub fn get_metadata(e: &Env) -> TokenMetadata {
+    e.storage().instance().get(&METADATA_KEY).unwrap_optimized()
+}
+
+pub fn set_metadata(e: &Env, metadata: &TokenMetadata) {
+    e.storage().instance().set(&METADATA_KEY, metadata);
+}
+
+//********** Persistent **********//
+
+// Total Supply
+
+pub fn get_total_supply(e: &Env) -> VotingUnits {
+    get_persistent_default(
+        e,
+        &TOTAL_SUPPLY_KEY,
+        || VotingUnits {
+            amount: 0,
+            timestamp: 0,
+        },
+        BALANCE_LIFETIME_THRESHOLD,
+        BALANCE_BUMP_AMOUNT,
+    )
+}
+
+pub fn set_total_supply(e: &Env, supply: &VotingUnits) {
+    e.storage().persistent().set(&TOTAL_SUPPLY_KEY, supply);
+}
+
+// Balance
+
+pub fn get_balance(e: &Env, address: &Address) -> i128 {
+    get_persistent_default(
+        e,
+        &DataKey::Balance(address.clone()),
+        || 0_i128,
+        BALANCE_LIFETIME_THRESHOLD,
+        BALANCE_BUMP_AMOUNT,
+    )
+}
+
+pub fn set_balance(e: &Env, address: &Address, balance: &i128) {
+    e.storage()
```

Comment:
We update TTL on `get` so we don't need to duplicate the update on read.

This does bring up a good point that the first time this is created it will not have a long TTL. This likely effects Blend too, so I'm gonna create an issue over there.

Reply:
This was a lie - we extend TTL when we call set in Blend. I'll do the same here in a follow up PR. Good catch!